pytest_localserver/https.py

```diff
@@ -111,13 +113,41 @@ class SecureContentServer(ContentServer):
     .. _pyOpenSSH: https://launchpad.net/pyopenssl
     """
 
-    def __init__(self, host="localhost", port=0, key=DEFAULT_CERTIFICATE, cert=DEFAULT_CERTIFICATE):
+    def __init__(self, host="localhost", port=0, key=DEFAULT_KEY, cert=DEFAULT_CERTIFICATE):
         """
         :param key: location of file containing the server private key.
         :param cert: location of file containing server certificate.
         """
 
-        super().__init__(host, port, ssl_context=(key, cert))
+        super().__init__(host, port, ssl_context=(cert, key))
+        self._cert = cert
+        
+    @property
+    def url(self):
+        # IMPORTANT:
+        # The supplied certificate is bound exclusively to "localhost" and does not include "127.0.0.1" in its Subject Alternative Name (SAN) field.
+        # As a result, validation against "127.0.0.1" will fail, even if the server's certificate is added to the trust chain.
+        # Current SAN configuration:
+        #  DNS:localhost
+        #
+        # To ensure successful validation when accessing the server via IP address, the certificate must be reissued with "127.0.0.1" included in the SAN field.
+        #
+        # The following command can be used for checking:
+        #       openssl x509 -in /path/to/certificate.pem -text -noout
+        host, port = self.server_address
+        return "https://localhost:%i" % port
```

Comment:
> I'm actually somewhat opposed to adding this method, because it's quite trivial for client code to construct the URL from `server_address`. I mean, that's basically why `server_address` exists.
> 
> Also, the thought has crossed my mind to add `url_for(path: str)`, like pytest-httpserver does, and that would clash with having a plain old `url()` method.

While I understand your point, I'd like to offer a counter perspective:
For consistency with the `httpserver`, it should have a `url` property, and this property should function as expected without any need for manual intervention. Ideally, we could just inherit the `url` property directly from the `httpserver`. However, it seems this approach is currently unfeasible due to issues with the SAN section of the certificate. I suspect the omission of `127.0.0.1` from the SAN was an oversight, given its relevance to the use cases for which the certificate is intended. Therefore, in the long run, regenerating the certificate to include this might make sense.

Reply:
Oh, yes I see what you mean. Somehow I missed that there was already a `url` property inherited from `WSGIServer`. (As you might know, I'm not the original author of most of this code, so I'm not as familiar with it as if it were my own project from the start.)

What would you think about making the default value of the `host` parameter be `localhost` in `WSGIServer` and `ContentServer` as well, for consistency with `SecureContentServer`? I don't think the current API makes any guarantees about specifically which hostname will be used, so changing from `127.0.0.1` to `localhost` would be consistent with the API spec. And in fact the documentation mentions `localhost`... if we just consistently use `localhost` everywhere, that should clear up any conflicts. (Unless @redtoad knows of a reason to use the IP address?)

We can still add `127.0.0.1` to the certificate SAN, but given that it's less common to use raw IP addresses with HTTPS, I think if we're going to standardize on anything as the default hostname, it should be `localhost`.